src/components/Tabs/components/TabList/TabList.tsx

```diff
@@ -0,0 +1,69 @@
+import { KeyboardEvent, MutableRefObject } from "react"
+import { Tab } from "../../types"
+import styles from "./TabList.module.css"
+
+interface TabListProps {
+  onClick: (tab: Tab) => void
+  onKeyDown: (event: KeyboardEvent) => void
+  tabs: Tab[]
+  refs: MutableRefObject<(HTMLLIElement | null)[]>
+}
+
+export function TabList({ tabs, onClick, onKeyDown, refs }: TabListProps) {
+  return (
+    <ul
+      aria-label="List of Tabs"
+      aria-orientation="horizontal"
+      className={styles.container}
+      role="tablist"
+    >
+      {tabs.map((tab) => (
+        <TabListItem
+          key={`tab:${tab.label}`}
+          onClick={onClick}
+          onKeyDown={onKeyDown}
+          tab={tab}
+          refs={refs}
+        />
+      ))}
+    </ul>
+  )
+}
+
+export interface TabListItemProps {
+  onClick: (tab: Tab) => void
+  onKeyDown: (event: KeyboardEvent) => void
+  tab: Tab
+  refs: MutableRefObject<(HTMLLIElement | null)[]>
+}
+
+export function TabListItem({
+  onClick,
+  onKeyDown,
+  tab,
+  refs,
+}: TabListItemProps) {
+  const className =
+    styles.item +
+    (tab.selected ? " " + styles.selected : "") +
+    (tab.disabled ? " " + styles.disabled : "")
+
+  const handleClick = () => onClick(tab)
```

Comment:
⛏️ nitpick alert - Is it necessary to pass in the whole tab object here? It looks like the only time you extract a property from this object is when you pass the `id` from the callback to your `onTabChange`. Seems like you could just pass the id here, unless you had a vision for future implementation.

Reply:
Nah, I think I was just hedging my bets. Passing the `id` makes most sense now. 👍